planner/core/find_best_task.go

```diff
@@ -423,7 +423,8 @@ func isCoveringIndex(columns []*expression.Column, indexColumns []*model.IndexCo
 		isIndexColumn := false
 		for _, indexCol := range indexColumns {
 			isFullLen := indexCol.Length == types.UnspecifiedLength || indexCol.Length == col.RetType.Flen
-			if col.ColName.L == indexCol.Name.L && isFullLen {
+			// We use col.OrigColName instead of col.ColName
+			if col.OrigColName.L == indexCol.Name.L && isFullLen {
```

Comment:
Is there other places that use `col.ColName.L` instead of `col.OrigColName.L`?

Reply:
I'm not sure.
It needs to be checked.

Reply:
Can we use `UniqueID` here for this case?

Reply:
No...  There's no place store the result converting `model.IndexInfo.Columns` to `expression.Columns`. So in many places we use `model.IndexColumn` directly compares with `expression.Column`